URBNSwiftCarousel/Classes/URBNSwCarouselTransitionController.swift

```diff
@@ -0,0 +1,89 @@
+    public func transitionDuration(transitionContext: UIViewControllerContextTransitioning?) -> NSTimeInterval {
+        return 0.5
+    }
+    
+    public func animateTransition(transitionContext: UIViewControllerContextTransitioning) {
+        
+        guard let fromVC = transitionContext.viewControllerForKey(UITransitionContextFromViewControllerKey),
+            toVC = transitionContext.viewControllerForKey(UITransitionContextToViewControllerKey)
+            else { return }
+        
+        let fromView = fromVC.view
+        let toView = toVC.view
+        
+        guard let topToVC = trueContextViewControllerFromContext(transitionContext, key: UITransitionContextToViewControllerKey) as? URBNSwCarouselTransitioning else { return }
+        
+        let duration = transitionDuration(transitionContext)
+        
+        UIView.animateWithDuration(duration, delay: 0, usingSpringWithDamping: springCompletionSpeed, initialSpringVelocity: 0, options: [], animations: {
+            fromView.alpha = 0.0
+            toView.alpha = 1.0
+            topToVC.configureAnimatingTransitionImageView?(self.transitionView)
+            
+            }) { (finished) in
+            transitionContext.completeTransition(!transitionContext.transitionWasCancelled())
+        }
+    }
+    
+    // MARK: UIViewControllerTransitioningDelegate methods
+    public func presentationControllerForPresentedViewController(presented: UIViewController, presentingViewController presenting: UIViewController, sourceViewController source: UIViewController) -> UIPresentationController? {
+        presentationController = URBNPresentationController(presentedViewController: presented, presentingViewController: source)
+        return presentationController
+    }
+    
+    public func animationControllerForPresentedController(presented: UIViewController, presentingController presenting: UIViewController, sourceController source: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        sourceViewController = source
+        return self
+    }
+    
+    public func animationControllerForDismissedController(dismissed: UIViewController) -> UIViewControllerAnimatedTransitioning? {
+        return self
+    }
+    
+    // MARK: Convenience
+    /** This method finds the true view controller for a given context's key - sometimes iOS will pick the rootview controller or navigation controller of a presenting view controller
+    */
+    public func trueContextViewControllerFromContext(transitionContext: UIViewControllerContextTransitioning, key: String) -> UIViewController? {
+        guard var vc = transitionContext.viewControllerForKey(key) else { return nil }
+        
+        if !vc.conformsToProtocol(URBNSwCarouselTransitioning) {
+            vc = sourceViewController
+        }
+        
+        if let topVC = vc.navigationController?.topViewController where vc .conformsToProtocol(URBNSwCarouselTransitioning) {
```

Comment:
[Dave Abrahms on optional protocol methods](http://thread.gmane.org/gmane.comp.lang.swift.evolution/13347)

I think it's worth thinking about this approach more. Why not provide a default implementation instead of making the method optional?